src/Sylius/Bundle/CoreBundle/Fixture/Factory/OrderExampleFactory.php

```diff
@@ -0,0 +1,282 @@
+
+            $product = $this->faker->randomElement($products);
+            $variant = $this->faker->randomElement($product->getVariants()->toArray());
+
+            $item->setVariant($variant);
+            $this->orderItemQuantityModifier->modify($item, random_int(1, 5));
+
+            $order->addItem($item);
+        }
+    }
+
+    protected function address(OrderInterface $order, string $countryCode): void
+    {
+        /** @var AddressInterface $address */
+        $address = $this->addressFactory->createNew();
+        $address->setFirstName($this->faker->firstName);
+        $address->setLastName($this->faker->lastName);
+        $address->setStreet($this->faker->streetAddress);
+        $address->setCountryCode($countryCode);
+        $address->setCity($this->faker->city);
+        $address->setPostcode($this->faker->postcode);
+
+        $order->setShippingAddress($address);
+        $order->setBillingAddress(clone $address);
+
+        $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_ADDRESS);
+    }
+
+    protected function selectShipping(OrderInterface $order): void
+    {
+        if (!$this->orderShippingMethodSelectionRequirementChecker->isShippingMethodSelectionRequired($order)) {
+            $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SKIP_SHIPPING);
+
+            return;
+        }
+
+        $channel = $order->getChannel();
+        $shippingMethods = $this->shippingMethodRepository->findEnabledForChannel($channel);
+        if (count($shippingMethods) === 0) {
+            throw new \InvalidArgumentException(sprintf(
+                'You have no shipping method available for the channel with code "%s", but they are required to proceed an order',
+                $channel->getCode()
+            ));
+        }
+
+        $shippingMethod = $this->faker->randomElement($shippingMethods);
+
+        /** @var ChannelInterface $channel */
+        $channel = $order->getChannel();
+        Assert::notNull($shippingMethod, $this->generateInvalidSkipMessage('shipping', $channel->getCode()));
+
+        foreach ($order->getShipments() as $shipment) {
+            $shipment->setMethod($shippingMethod);
+        }
+
+        $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
+    }
+
+    private function selectPayment(OrderInterface $order): void
+    {
+        if (!$this->orderPaymentMethodSelectionRequirementChecker->isPaymentMethodSelectionRequired($order)) {
+            $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SKIP_PAYMENT);
+            return;
+        }
+        $paymentMethod = $this
+            ->faker
+            ->randomElement($this->paymentMethodRepository->findEnabledForChannel($order->getChannel()))
+        ;
+
+        /** @var ChannelInterface $channel */
+        $channel = $order->getChannel();
```

Comment:
This value could be assigned before the creation of `$paymentMethod`, so it can be used in `this->paymentMethodRepository->findEnabledForChannel($channel)`